utils/s3/retryable_http_client.hh

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#pragma once
+#include "retry_strategy.hh"
+#include <seastar/http/client.hh>
+
+namespace aws {
+
+class retryable_http_client {
+public:
+    using error_handler = std::function<void(std::exception_ptr)>;
+
+    retryable_http_client(std::unique_ptr<seastar::http::experimental::connection_factory>&& factory,
+                          unsigned max_conn,
+                          error_handler error_func,
```

Comment:
Moving client into own file and equipping it with extra functionality in one change is hard to review. Please, split